decidim-admin/app/controllers/decidim/admin/newsletters_controller.rb

```diff
@@ -0,0 +1,111 @@
+      end
+
+      def new
+        authorize! :create, Newsletter
+        @form = form(NewsletterForm).instance
+      end
+
+      def show
+        @newsletter = base_query.find(params[:id])
+        email = NewsletterMailer.newsletter(current_user, @newsletter)
+        @email_subject = email.subject
+        @email_body = Nokogiri::HTML(email.body.decoded).css("table.container").to_s
+        authorize! :read, @newsletter
+      end
+
+      def create
+        authorize! :create, Newsletter
+        @form = form(NewsletterForm).from_params(params)
+
+        CreateNewsletter.call(@form, current_user) do
+          on(:ok) do |newsletter|
+            flash.now[:notice] = I18n.t("newsletters.create.success", scope: "decidim.admin")
+            redirect_to action: :show, id: newsletter.id
+          end
+
+          on(:invalid) do |newsletter|
+            @newsletter = newsletter
+            flash.now[:error] = I18n.t("newsletters.create.error", scope: "decidim.admin")
+            render action: :new
+          end
+        end
+      end
+
+      def edit
+        @newsletter = base_query.find(params[:id])
+        authorize! :update, @newsletter
+        @form = form(NewsletterForm).from_model(@newsletter)
+      end
+
+      def update
+        @newsletter = base_query.find(params[:id])
+        authorize! :update, Newsletter
+        @form = form(NewsletterForm).from_params(params)
+
+        UpdateNewsletter.call(@newsletter, @form, current_user) do
+          on(:ok) do |newsletter|
+            flash.now[:notice] = I18n.t("newsletters.update.success", scope: "decidim.admin")
+            redirect_to action: :show, id: newsletter.id
+          end
+
+          on(:invalid) do |newsletter|
+            @newsletter = newsletter
+            flash.now[:error] = I18n.t("newsletters.update.error", scope: "decidim.admin")
+            render action: :edit
+          end
+        end
+      end
+
+      def destroy
+        @newsletter = base_query.find(params[:id])
+        authorize! :destroy, @newsletter
+
+        if @newsletter.sent?
+          flash.now[:error] = I18n.t("newsletters.destroy.error_already_sent", scope: "decidim.admin")
+          redirect_to :back
+        else
+          @newsletter.destroy!
+          flash[:notice] = I18n.t("newsletters.destroy.success", scope: "decidim.admin")
+          redirect_to action: :index
+        end
+      end
+
+      def deliver
+        @newsletter = base_query.find(params[:id])
+        authorize! :update, @newsletter
+
+        DeliverNewsletter.call(@newsletter) do
+          on(:ok) do
+            flash[:notice] = I18n.t("newsletters.deliver.success", scope: "decidim.admin")
+            redirect_to action: :index
+          end
+
+          on(:invalid) do
+            flash[:error] = I18n.t("newsletters.deliver.error", scope: "decidim.admin")
+            redirect_to action: :show
+          end
+        end
+      end
+
+      private
+
+      def collection
```

Comment:
`collection` is used just once and it's wrong. Why not use `base_query` everywhere and rename it to collection, since base_query is weird?